ludwig/api.py

```diff
@@ -269,11 +269,11 @@ def __init__(
         self,
         config: Union[str, dict],
         logging_level: int = logging.ERROR,
-        backend: Union[Backend, str] = None,
-        gpus: Union[str, int, List[int]] = None,
+        backend: Union[Backend, str, None] = None,
+        gpus: Union[str, int, List[int], None] = None,
         gpu_memory_limit: Optional[float] = None,
         allow_parallel_threads: bool = True,
-        callbacks: List[Callback] = None,
+        callbacks: Union[List[Callback], None] = None,
```

Comment:
Nit: we tend to use `Optional[List[Callback]]` for params that accept `None` as a valid value. I think it makes sense to use `Union` when there are already a `Union` over multiple types, but `Optional` would be more consistent with the rest of the codebase for situations like this where there is only one valid type that can optionally be `None`. What do you think?

Reply:
@tgaddair I will change it to `Optional` -- totally understood in terms of supporting the presently established coding pattern.  Having said that, I would like to suggest something entirely different for the future.  If we place `from __future__ import annotations` at the top of the module, we will be able to use the `|` operator, which is a shorthand for `Union`.  That way, we will be able to differentiate the true optionality (i.e., argument can be legitimately omitted, without a default needed to be supplied) from the actual possibility for a parameter to be `None` or other values.  What are your thoughts about this?  Thank you very much.  P.S.: I will notify you once I have made the change to `Optional` -- thanks.

Reply:
@tgaddair I made the changes -- please re-review at your convenience.  Thank you.

Reply:
Thanks! Totally open to changing the convention if using `|` is the preferred notation going forward. It makes sense to me that it's worth distinguishing between an optional kwarg and something that can be explicitly set to `None`. Do you think it would be better to do this now, or wait until 3.10 (which I believe is the first version to support this officially) is the min supported version? I'm pretty open either way.